cogs/ahk/ahk.py

```diff
@@ -295,24 +296,50 @@ async def cloudahk_call(self, ctx, code, lang='ahk'):
 		stdout, time = result['stdout'].strip(), result['time']
 
 		stdout = stdout.replace('`', '`\u200b')
+		file = None
+		encoded_out = stdout.encode('utf-8')
+		if len(stdout) < 1800 and stdout.count('\n') < 20 and stdout.count('\r') < 20:
+			#upload as plaintext
+			valid_response = 'No output.\n' if stdout == '' else '\n```autoit\n{0}\n```'.format(stdout)
+		elif len(encoded_out) < 100000 and AHK_GUILD_ID == ctx.guild.id:
+			#upload to ahkscript.org
+			data = {'name': 'Ace Bot (Clone)', 'code': stdout}
+			async with ctx.http.post('https://p.ahkscript.org/',data=data) as resp:
+				if resp.status != 200:
+					raise commands.CommandError('Pastebin failed.')
+				valid_response = ' No output.\n' if stdout == '' else f' Results too large. Pasted. {resp.url}\n'
+		elif len(encoded_out) < (1000*1000*4000): # should be 4mb
+			fp = io.BytesIO(stdout.encode('utf-8'))
+			file = discord.File(fp, 'results.txt')
+			valid_response = ' No output.\n' if stdout == '' else ' Results too large. See attached file.\n'
+		else:
+			raise commands.CommandError('Output greater than 4mb.')
 
-		if len(stdout) > 1800 or stdout.count('\n') > 12:
-			raise commands.CommandError('Output too large.')
-
-		out = '{0}{1}`Processing time: {2}`'.format(
+		out = '{}{}{}'.format(
 			ctx.author.mention,
-			' No output.\n' if stdout == '' else '\n```autoit\n{0}\n```'.format(stdout),
-			'Timed out' if time is None else '{0:.1f} seconds'.format(time),
+			valid_response,
+			'`Processing time: {}`'.format('Timed out' if time is None else '{0:.1f} seconds'.format(time))
 		)
+		
 
-		await ctx.send(out)
+		try:
+			await ctx.reply(content=out, file=file)
+		except discord.HTTPException:
+			await ctx.send(content=out, file=file)
 
 		return stdout, time
 
 	@commands.command()
 	@commands.cooldown(rate=1.0, per=5.0, type=commands.BucketType.user)
 	async def ahk(self, ctx, *, code):
 		'''Run AHK code through CloudAHK. Example: `ahk print("hello world!")`'''
+		
+		if code.startswith('https://p.ahkscript.org/'):
+			url = code.replace('?p=','?r=')
+			async with ctx.http.get(url) as resp:
+				if resp.status == 200 and str(resp.url) == url:
+					code = await resp.text()
+
 
 		stdout, time = await self.cloudahk_call(ctx, code)
```

Comment:
This will be called even if the API call above fails.

Reply:
That _was_ intentional because if its not a valid link, assume that the user wanted to run that as code.